# Motility


## Bacillus TODO list
















### Dry


*  Electronic interaction graph - cytoscape, ONDEX, graphviz, ... ?

*  Image analysis of microscopy data

*  Look at Neil's B. Subtilis integrated functional interaction network - links to yjbM or ywaC from sigD?

*  Work on JWV's sporulation model


















































## Strains

### Stock strains

 | Strain                                                                                                                                                                                                                                                                                                                                                     | Genotype | Relevant genotype | Publication | Comment|
 | ------                                                                                                                                                                                                                                                                                                                                                     | -------- | ----------------- | ----------- | --------
 | DS901|PY79 [swrA](http://www.uniprot.org/uniprot/O32266)(PY79) (sfp0) [amyE](http://www.uniprot.org/uniprot/P00691)::P[hag](http://www.uniprot.org/uniprot/P02968)-gfp Cm|PY79 P[hag](http://www.uniprot.org/uniprot/P02968)-gfp| [Kearns and Losick (05)](http://dx.doi.org/10.1101/gad.1373905) |from [Remco Kort](http://staff.science.uva.nl/~rkort/)|
 | LH151|168 amyE::Phag-gfp Cm |168 Phag-gfp| |LH derived from DS901|                                                                                                                                                                                                                                                                                        
 | LH153| 168 relA::ery | 168 relAΔ | | 168xTW20 |                                                                                                                                                                                                                                                                                                          
 | LH154|PY79 swrA(PY79) ([sfp](http://www.uniprot.org/uniprot/P39135)0) amyE::Phag-gfp Cm [relA](http://www.uniprot.org/uniprot/O54408)::Ery |PY79 [relA](http://www.uniprot.org/uniprot/O54408)Δ Phag-gfp| |Derived from DS901|                                                                                                                           
 | LH155|168 amyE::Phag-gfp Cm relA::Ery |168 relAΔ Phag-gfp| |Derived from LH151|                                                                                                                                                                                                                                                                          
 | FlhF |168 ([trpC](http://www.uniprot.org/uniprot/P03964)2) [flhF](http://www.uniprot.org/uniprot/Q01960)::cat CmR|168 [flhF](http://www.uniprot.org/uniprot/Q01960)Δ|[Zanen el al (04)](http://www.ncbi.nlm.nih.gov/pubmed/15317803)|from van Dijl|                                                                                                      
 | BFA2616|168 (trpC2) ylxH::pMutin2 EmR|168 [ylxH](http://www.uniprot.org/uniprot/P40742)Δ Pspac-[cheB](http://www.uniprot.org/uniprot/Q05522)|[Zanen el al (04)](http://www.ncbi.nlm.nih.gov/pubmed/15317803)|from van Dijl|                                                                                                                              
 | TW20|relA::mls amyE::spacp-relA-cat|Pspac-relA|[Wendrich & Marahiel (03)](http://dx.doi.org/10.1046/j.1365-2958.1997.5511919.x)|from Marburg, Germany|                                                                                                                                                                                                    
 | DegUhy | 168, P[aprE](http://www.uniprot.org/uniprot/P04189)-gfp, Cmr, amyE::Phyper-spank-degUhy, Spr |168 PaprE-gfp Phy-spank-[degU](http://www.uniprot.org/uniprot/P13800)|[Veening et al (08)](http://dx.doi.org/10.1038/msb.2008.18) |from JWV |                                                                                                      
 | PS258 | [codY](http://www.uniprot.org/uniprot/P39779)::ery | [codY](http://www.uniprot.org/uniprot/P39779)Δ | | from Imke |                                                                                                                                                                                                                              
 | RIK908|trpC2 ywaC::spc|[ywaC](http://www.uniprot.org/uniprot/P39583)Δ|[Nanamiya et al (08)](http://www.ncbi.nlm.nih.gov/pubmed/18067544) |from Nanamiya|                                                                                                                                                                                                 
 | RIK909|trpC2 yjbM::cat|[yjbM](http://genolist.pasteur.fr/SubtiList/genome.cgi?gene_detail+BG13142)Δ|[Nanamiya et al (08)](http://www.ncbi.nlm.nih.gov/pubmed/18067544) |from Nanamiya|                                                                                                                                                                   
 | RIK1003|trpC2 relA::erm ΔyjbM ywaC::spc|relAΔ yjbMΔ ywaCΔ|[Nanamiya et al (08)](http://www.ncbi.nlm.nih.gov/pubmed/18067544) |from Nanamiya|                                                                                                                                                                                                          
 | DNA-RIK900|trpC2 relA::erm|relAΔ|[Nanamiya et al (08)](http://www.ncbi.nlm.nih.gov/pubmed/18067544) |from Nanamiya|                                                                                                                                                                                                                                      
 | hag-gfp | 168, Phag-gfp Cmr|168 Phag-gfp| [Lulko et al (07)](http://dx.doi.org/10.1128/AEM.00472-07)| from JWV |                                                                                                                                                                                                                                          
 | LH156| [amyE](http://www.uniprot.org/uniprot/P00691)::P[rpsF](http://www.uniprot.org/uniprot/P21468)-short-gfp, Spec | P[rpsF](http://www.uniprot.org/uniprot/P21468)-short-gfp | | Henrik/Ugoo |                                                                                                                                                         
 | LH157| amyE::PrpsF-long-gfp, Spec |PrpsF-long-gfp | | Henrik/Ugoo |                                                                                                                                                                                                                                                                                       
 | LH158| amyE::P[rpsD](http://www.uniprot.org/uniprot/P21466)-gfp, Spec |P[rpsD](http://www.uniprot.org/uniprot/P21466)-gfp | | Henrik/Ugoo |                                                                                                                                                                                                               
 | JW1| amyE::P[rrnB](http://subtiwiki.uni-goettingen.de/wiki/index.php/RrnB-16S)-gfp-[ssrA](http://www.subtiwiki.uni-goettingen.de/wiki/index.php/SsrA), CmR | P[rrnB](http://subtiwiki.uni-goettingen.de/wiki/index.php/RrnB-16S)-gfp-[ssrA](http://www.subtiwiki.uni-goettingen.de/wiki/index.php/SsrA)| | J-WV |                                         
 | LH159| amyE::P[rpsJ](http://www.uniprot.org/uniprot/P21471)-gfp |P[rpsJ](http://www.uniprot.org/uniprot/P21471)-gfp | | Henrik/Ugoo |                                                                                                                                                                                                                     






### My strains

 | Strain                                                                                                                              | Genotype | Relevant genotype | Origin | Comment|
 | ------                                                                                                                              | -------- | ----------------- | ------ | --------
 | a|PY79 amyE::Phag-gfp Cm ywaC::spc |PY79 ywaCΔ Phag-gfp|DS901xRIK908| |                                                           
 | b|168 amyE::Phag-gfp Cm ywaC::spc|168 ywaCΔ Phag-gfp|LH151xRIK908| |                                                              
 | c|PY79 amyE::Phag-gfp Cm relA::Ery ywaC::spc|PY79 relAΔ ywaCΔ Phag-gfp|LH154xRIK908| |                                           
 | d|168 Phag-gfp Cm ywaC::spc|168 ywaCΔ Phag-gfp|hag-gfpxRIK908| |                                                                  
 | e|PY79 amyE::Phag-gfp Cm ylxH::pMutin2 EmR|PY79 Phag-gfp ylxHΔ Pspac-cheB|DS901xBFA2616|was a poor transformation, but seems fine|
 | f|168 amyE::Phag-gfp Cm ylxH::pMutin2 EmR|168 Phag-gfp ylxHΔ Pspac-cheB|LH151xBFA2616|have two versions - both seem fine|         
 | g|168 amyE::Phag-gfp Cm ywaC::spc relA::Ery|168 Phag-gfp relAΔ ywaCΔ|(b)xLH153| |                                                
 | h*|168 amyE::Phag-gfp Cm relA::Ery ywaC::spc|168 Phag-gfp relAΔ ywaCΔ|LH155xRIK908|failed - low priority - use g instead|        
 | i|168 yjbM::cat ywaC::spc|168 yjbMΔ ywaCΔ|RIK909xRIK908|was a poor transformation |                                              
 | j|168 amyE::Phag-gfp Cm codY::Ery|168 Phag-gfp codYΔ|LH151xPS258| |                                                               
 | k*|PY79 amyE::Phag-gfp Cm ylxH::pMutin2 EmR|PY79 Phag-gfp ylxHΔ Pspac-cheB |DS901xBFA2616|same as e!|                             
 | l*|PY79 amyE::Phag-gfp Cm codY::Ery|PY79 Phag-gfp codYΔ |DS901xPS258 |not yet attempted - low priority|                           
 | m|168 Phag-gfp Cmr ylxH::pMutin2 EmR|168 Phag-gfp ylxHΔ Pspac-cheB |hag-gfpxBFA2616| |                                            
 | n|168 Phag-gfp Cmr codY::Ery|168 Phag-gfp codYΔ |hag-gfpxPS258|grows slowly on linc/erm plates|                                   
 | o|168 amyE::Phyper-spank-degUhy Spr|168 Phy-spank-[degU](http://www.uniprot.org/uniprot/P13800)|168xDegUhy| |                      
 | p|168 amyE::Phyper-spank-degUhy Spr Phag-gfp Cmr|168 Phag-gfp Phy-spank-degU |hag-gfp x o| |                                       
 | q*|168 amyE::Phyper-spank-degUhy Spr Phag-gfp Cmr|168 Phag-gfp Phy-spank-degU  |o x hag-gfp|low priority - use p instead|          
 | r|168 amyE::Phag-gfp Nm |168 Phag-gfp |LH151xECE73(pCm::Nm)| |                                                                     
 | s|168 Phag-gfp Nm |168 Phag-gfp|hag-gfpxECE73(pCm::Nm)| |                                                                          
 | t*|168 yjbM::cat ywaC::spc relA::ery |168 yjbMΔ ywaCΔ relAΔ |i x LH153|not yet attempted - low priority|                        
 | u|168 Phag-gfp Nm  flhF::cat|168 Phag-gfp flhFΔ |s x FlhF| |                                                                      
 | v|168 Phag-gfp Nm  yjbM::cat|168 Phag-gfp yjbMΔ |s x RIK909| |                                                                    
 | w*|168 Phag-gfp Nm  yjbM::cat ywaC::spc|168 Phag-gfp yjbMΔ ywaCΔ |i x s|not yet attempted|                                       
 | y*|168 Phag-gfp Nm yjbM::cat ywaC::spc relA::ery |168 Phag-gfp yjbMΔ ywaCΔ relAΔ |w x LH153|not yet attempted|                  
 | z*|168 Phag-gfp flhF::cat ylxH::pMutin2 EmR |168 Phag-gfp flhFΔ ylxHΔ Pspac-cheB |u x BFA2616|not yet attempted|                 
 | aa| 168 Phag-gfp Cm | 168 Phag-gfp | 168 x hag-gfp | |                                                                             
 | ab| 168 amyE::Phag-gfp Nm flhF::cat | 168 Phag-gfp flhFΔ | r x FlhF | |                                                           
 | ac| 168 amyE::Phag-gfp Nm yjbM::cat | 168 Phag-gfp yjbMΔ | r x RIK909 | |                                                         
 | ad*| 168 amyE::Phag-gfp Nm flhF::cat ylxH::pMutin2 EmR | 168 Phag-gfp flhFΔ  ylxHΔ Pspac-cheB| ab x BFA2616 |SS made|            
 | ae*| 168 amyE::Phag-gfp Nm yjbM::cat ywaC::spc | 168 Phag-gfp yjbMΔ ywaCΔ | ac x RIK908 |SS made|                                
 | af*| 168 amyE::Phag-gfp Nm yjbM::cat ywaC::spc relA::Ery| 168 Phag-gfp yjbMΔ ywaCΔ  relAΔ| ae x LH153 |SS made|                 
 | ag| 168 amyE::Phag-gfp Nm relA::ery | 168 Phag-gfp relAΔ | r x LH153 | |                                                          
 | ah*| 168 Phag-gfp Nm relA::ery | 168 Phag-gfp relAΔ | s x LH153 |SS made|                                                         
 | ai*| 168 Phag-gfp Nm relA::ery amyE::spacp-relA-cat| 168 Phag-gfp relAΔ Pspac-relA| ah x TW20 |SS made|                           
 | aj*| 168 amyE::Phag-gfp Nm yjbM::cat relA::ery | 168 Phag-gfp yjbMΔ relAΔ  | ac x LH153 | ss attempting|                         
 | ak*| 168 atpA::pMutinC1 Ery P`<sub>`hag`</sub>`-gfp cat | 168 atpAΔ Erm P`<sub>`hag`</sub>`-gfp Cm | atpA x hag-gfp | ss attempting|      

* - under construction...


### Other interesting strains

 | Strain                                                                                                                                                                                               | Genotype | Relevant genotype | Publication | Comment|
 | ------                                                                                                                                                                                               | -------- | ----------------- | ----------- | --------
 | PB5368 | Pspac inducible fla/che| |[Amati et al (04)](http://www.ncbi.nlm.nih.gov/pubmed/15342569)|not got|                                                                                         
 | W1111 | amyE::flgB P(A)-gfp (erm)| [flgB](http://www.uniprot.org/uniprot/P24500) P(A)-gfp|[Kobayashi (07)](http://dx.doi.org/10.1111/j.1365-2958.2007.05923.x) | not got - don't want to alert him |






## Experimental data
























### Microscopy


*  **2009-01-13** Looked at strains DS901 and LH151 at various time points. First day on the microscope, so a few problems... Have images {{hetero:bs-135-2009-01-13-sep.pdf|separately}} and {{hetero:bs-135-2009-01-13.pdf|overlaid}}.

*  **2009-01-14** Looked at strains DS901, LH151 and LH155 at various time points. Have images {{hetero:bs-135-2009-01-14-sep.pdf|separately}} and {{hetero:bs-135-2009-01-14.pdf|overlaid}}.

*  **2009-01-22** Looked at strains DS901, LH151, LH154, hag-gfp. Have images {{hetero:bs-135-2009-01-22-sep2.pdf|separately}} and {{hetero:bs-135-2009-01-22-2.pdf|overlaid}}. Includes first attempt at DAPI staining, though not especially successful. 

*  **2009-02-04** Looked at strains DS901, LH151, LH155, hag-gfp, (a), (b), (c), (d). Have images {{hetero:bs-135-2009-02-04-sep2.pdf|separately}} and {{hetero:bs-135-2009-02-04.pdf|overlaid}}. DAPI stained again, but again not great.

*  **2009-02-19** Looked at strains DS901, LH151, (a), (c). Have images {{hetero:bs-135-2009-02-19-sep.pdf|separately}} and {{hetero:bs-135-2009-02-19.pdf|overlaid}}. Didn't have time for DAPI.

*  **2009-02-25** Looked at strains LH151, and (f), with and without IPTG (also looked at DS901 and (e) with and without IPTG, but no pictures due to poor agarose slide). Just uploaded {{hetero:bs-135-2009-02-25-sep.pdf|images separately}}. Just one time point, around 6 hours after diluting (200x). It is clear that for the strains with the inducible promotor, without IPTG they are long and filamentous with little gfp expression, whereas with IPTG, the cells are short, and have relatively high levels of gfp.

*  **2009-03-10** Looked at LH151, LH155, g, j, f-, f+, p-, p+ at 2 time points. Time point one was about 3 hours, time point 2 was about 5 hours after dilution. Have images {{hetero:bs-135-2009-03-10-sep.pdf|separately}} and {{hetero:bs-135-2009-03-10.pdf|overlaid}}.

*  **2009-05-13** Looked at some **ribosomal** strains as well as some hag strains: LH156, LH157, LH158, JW1, DS901, LH151, hag-gfp, LH151(MMC). {{hetero:bs-135-2009-05-13.pdf|Images}} produced using new python overlay scripts. t1 around 5 hours and t2 around 6 hours (no early time points)

*  **2009-05-14** Looked at DS901, LH151, hag-gfp, LH155 and g. {{hetero:bs-135-2009-05-14.pdf|Images}}. Also attempted a {{hetero:bs-135-2009-05-14-tl.avi|short time-lapse}} using the 135. Not very successful, but mainly due to the fact that autofocus doesn't seem to work. 

### Flow cytometry


*  **2009-02-03** Looked at strains DS901, LH151, LH155, hag-gfp, (a), (b), (c), (d). Some basic {{hetero:bs-fc-2009-02-03.pdf|flowViz plots}} show the essential patterns of heterogeneity. The analysis reveals mainly that the experiment didn't work especially well. The samples at the first time point were too dilute, and at the second time point there was still little evidence of bimodality. This doesn't really agree with the results of the microscopy however, so really need to have another go at this. The overnight cultures worked best, with bimodal distributions for (a), (b), and LH151. There was also a broad range of fluorescence for (d). The other 4 strains: (c), DS901, LH155 and hag-gfp, showed little evidence of bimodality, which again somewhat contradicts the microscopy. However, the results do suggest to some extent that the relA mutants (c) and LH155 are less heterogeneous, which is interesting. JWV thinks that the long filamentous cells in strains like DS901 don't really fly properly in the flow cytometer, which could also explain some of the results.

*  **2009-02-19** Looked at strains DS901, LH151, (a), and (c). {{hetero:bs-fc-2009-02-19.pdf|FlowViz plots}}.

*  **2009-02-25** Looked at strains DS901, LH151, and (e) and (f) with and without IPTG. {{hetero:bs-fc-2009-02-25.pdf|FlowViz plots}}. t1 is around 6 hours and t2 around 7.5 hours after dilution. t1 should be viewed with caution, as I ran the machine at a speed of 4 instead of 0.4.

*  **2009-03-10** Looked at same strains as for the microscope. {{hetero:bs-fc-2009-03-10.pdf|Flowviz plots}}. t1 is around 4 hours and t2 is around 6 hours.

*  **2009-04-08** Looked at DS901, LH151, hag-gfp, LH155, m-IPTG, m+IPTG (1mM). Two time points. 2nd time point has cells spun down and re-suspended in PBS. t1 around 4.5hrs and t2 around 6hrs. {{hetero:bs-fc-2009-04-08.pdf|Flowviz plots}}.
    * Also did a dilution experiment at around 5 hours with LH151 and different dilutions of the cells in LB and different amounts of LB in the PBS. {{hetero:bs-fc-2009-04-08-dil.pdf|Flowviz plots}}. Dx is the cell density in LB, with D0 being neat and Dx having a dilution of 10^x relative to neat. The other number is the amount of this LB culture in ul added to 1ml PBS.

*  **2009-05-12** Dilution experiment using cells pelleted and re-suspended in PBS. Looks promising. Looks like adding 100ul of the re-suspended cells gives a good signal without much background. {{hetero:bs-fc-2009-05-12.pdf|Flowviz plots}}.

*  **2009-05-13** Strains DS901, LH151, hag-gfp (grown in LB) and LH151 grown in CMM. Using new protocol which eliminates most of the background. Looks interesting, but substantial differences between the strains... {{hetero:bs-fc-2009-05-13.pdf|Flowviz plots}}.

*  **2009-05-27** Strains LH151, hag-gfp, r, s, b, u, LH155 and g. t1=2 hrs, t2=3hrs, t3=5hrs, t4=6hrs. Lots of information and heterogeneity, but no bi-modality... {{hetero:bs-fc-2009-05-27.pdf|Flowviz plots}}.

*  **2009-06-03** Strains DS901, LH151, hag-gfp, s, r.1, r.2, s.3 (for final selection of r/s strains to keep). Diluted 200x in LB. t1 is 2 hrs, t2 is 3.5 hrs. {{hetero:bs-fc-2009-06-03.pdf|Flowviz plots}}. 

*  **2009-06-17** Strains LH151, LH155, b, g, j, ab, ac. Diluted 100x in LB (except g which was only 33x). t1 is 3 hrs and t2 is 4 hrs. {{hetero:bs-fc-2009-06-17.pdf|Flowviz plots}}. t1 samples were accidentally run at a speed of 4 rather than 0.4 (as was the first sample at t2, but this was repeated at the correct speed).

*  **2009-07-08** Best data yet! Strains: r, s, aa, ab, ac, ad, ae, af, ag. Nice early time point (2 hours), spun down 2ml, and cells processed on ice. Grown in LB at 37 after diluting 100x. This data shows lots of heterogeneity and even some bimodality. Data confirms lots of stuff I've observed informally under the microscope. In particular, it confirms that LH151 is a bit more heterogenous than JW's hag-gfp. It also shows that hag-gfp back-crossed to LH's 168 is much more heterogeneous than either LH151 or hag-gfp. I do wonder if that strain would form a much better starting point than either LH151 or hag-gfp for introducing additional mutations... flhF doesn't seem to have much effect on anything. It is the relA and relA like mutants that are most interesting. Deleting yjbM leads to much greater heterogeneity in both size and fluorescence, with no correlation between the two. There is bimodality in the fluorescence distribution, with the shorter cells separating into separate "on" and "off" populations. Additionally deleting ywaC leads to loss of the short bright cells, giving an overall picture not too dissimilar to wild type. Deletion of relA leads generally to cells that are shorter and brighter than wild type. However, there is considerable (greater) heterogeneity in both size and fluorescence, and bimodality in the fluorescece distribution, with the shorter cells separating quite cleanly into an "on" (the majority) and "off" population. However, additionally deleting yjbM and ywac gives a distribution again not too dissimilar to wild type. Deletion of either yjbM or relA leads to a quite similar distribution - the main difference being that the relA mutants are much more strongly biased towards the "on" state, and the yjbM mutants being much more evenly spit between "on" and "off". {{hetero:bs-fc-2009-07-08.pdf|Flowviz plots}}.


















### Time-lapse microscopy


*  **2009-04-22** Looked at DS901 and LH151 (with lots of help from JWV). 4 movies - 2 frames for each strain. Cultures diluted down, but neat LB agarose probably too rich - need to look at less rich media. {{hetero:dv-2009-04-21_ds901_lh151_01_1_r3d-merge.avi|Movie 1}}, {{hetero:dv-2009-04-21_ds901_lh151_01_2_r3d-merge.avi|movie 2}}, {{hetero:dv-2009-04-21_ds901_lh151_01_3_r3d-merge.avi|movie 3}}, {{hetero:dv-2009-04-21_ds901_lh151_01_4_r3d-merge.avi|movie 4}}. These are low-res compressed AVI files. Still have the raw Tiff stacks. I think 1 and 2 are DS901, but not absolutely sure...

*  **2009-04-24** My first attempt at using the DeltaVision myself. Again it is DS901 and LH151, but I've no idea which is which, and it hardly seems to matter. Still much to learn... {{hetero:dv-djw-test05_1_r3d-merge.avi|Movie 1}}, {{hetero:dv-djw-test05_2_r3d-merge.avi|movie 2}}, {{hetero:dv-djw-test05_3_r3d-merge.avi|movie 3}}, {{hetero:dv-djw-test05_4_r3d-merge.avi|movie 4}}.

*  **2009-05-14** Short time-lapse with 15% LB in 1.5% regular agar using DS901 and LH151 again. Movies 1 and 2 are DS901 and movies 3 and 4 are LH151. {{hetero:dv-2009-05-14_1_r3d-merge.avi|Movie 1}}, {{hetero:dv-2009-05-14_2_r3d-merge.avi|movie 2}}, {{hetero:dv-2009-05-14_3_r3d-merge.avi|movie 3}}, {{hetero:dv-2009-05-14_4_r3d-merge.avi|movie 4}}.

*  **2009-05-20** Overnight time-lapse with strains r (1, 2, and 3) and s (4, 5, 6) growing on CMM with 1.5% low gelling temp agar. 10 mins per frame. {{hetero:dv-2009-05-20-rs01_1_r3d-merge.avi|Movie 1}}, {{hetero:dv-2009-05-20-rs01_2_r3d-merge.avi|movie 2}}, {{hetero:dv-2009-05-20-rs01_3_r3d-merge.avi|movie 3}}, {{hetero:dv-2009-05-20-rs01_4_r3d-merge.avi|movie 4}}, {{hetero:dv-2009-05-20-rs01_5_r3d-merge.avi|movie 5}}, {{hetero:dv-2009-05-20-rs01_6_r3d-merge.avi|movie 6}}. Cells stop growing after a few hours and gradually start to shrink and die.

*  **2009-06-03** Overnight time-lapse with strains DS901 (1, 2, 3) and LH151 (4, 5, 6) growing on 15% LB with 1.5% LGT agar. 10 mins per frame. {{hetero:dv-2009-06-03_1_r3d-merge.avi|Movie 1}}, {{hetero:dv-2009-06-03_2_r3d-merge.avi|movie 2}}, {{hetero:dv-2009-06-03_3_r3d-merge.avi|movie 3}}, {{hetero:dv-2009-06-03_4_r3d-merge.avi|movie 4}}, {{hetero:dv-2009-06-03_5_r3d-merge.avi|movie 5}}, {{hetero:dv-2009-06-03_6_r3d-merge.avi|movie 6}}. 

## Interactions

First attempt at an electronic {{hetero:bs-motility-v1.cys|interaction network}} (requires [cytoscape](http://www.cytoscape.org/))
















## Background/Literature


*  fla/che operon. Genes: [flgB](http://www.uniprot.org/uniprot/P24500), [flgC](http://www.uniprot.org/uniprot/P24501), [fliE](http://www.uniprot.org/uniprot/P24502), [fliF](http://www.uniprot.org/uniprot/P23447), [fliG](http://www.uniprot.org/uniprot/P23448), [fliH](http://www.uniprot.org/uniprot/P23449), [fliI](http://www.uniprot.org/uniprot/P23445), [fliJ](http://www.uniprot.org/uniprot/P20487), [ylxF](http://www.uniprot.org/uniprot/P23454), [fliK](http://www.uniprot.org/uniprot/P23451), [ylxG](http://www.uniprot.org/uniprot/P23455), [flgE](http://www.uniprot.org/uniprot/P23446), [fliL](http://www.uniprot.org/uniprot/P23452), [fliM](http://www.uniprot.org/uniprot/P23453), [fliY](http://www.uniprot.org/uniprot/P24073), [cheY](http://www.uniprot.org/uniprot/P24072), [fliZ](http://www.uniprot.org/uniprot/P35536), [fliP](http://www.uniprot.org/uniprot/P35528), [fliQ](http://www.uniprot.org/uniprot/P35535), [fliR](http://www.uniprot.org/uniprot/P35537), [flhB](http://www.uniprot.org/uniprot/P35538), [flhA](http://www.uniprot.org/uniprot/P35620), [flhF](http://www.uniprot.org/uniprot/Q01960), [ylxH](http://www.uniprot.org/uniprot/P40742), [cheB](http://www.uniprot.org/uniprot/Q05522), [cheA](http://www.uniprot.org/uniprot/P29072), [cheW](http://www.uniprot.org/uniprot/P39802), [cheC](http://www.uniprot.org/uniprot/P40403), [cheD](http://www.uniprot.org/uniprot/P40404), [sigD](http://www.uniprot.org/uniprot/P10726), [swrB](http://www.uniprot.org/uniprot/P40405). Kearns and Losick (05)

*  [sigA](http://www.uniprot.org/uniprot/P06224), [swrA](http://www.uniprot.org/uniprot/O32266), [sigD](http://www.uniprot.org/uniprot/P10726), [swrB](http://www.uniprot.org/uniprot/P40405) all important regulators of fla/che, etc.

*  GTPases: [Ffh](http://www.uniprot.org/uniprot/P37105), [FtsY](http://www.uniprot.org/uniprot/P51835), and [FlhF](http://www.uniprot.org/uniprot/Q01960) - the latter is dispensible for protein secretion (Zanen et al, 04). Other GTPases, including [FtsZ](http://www.uniprot.org/uniprot/P17865)...

*  Alarmone synthetases: [relA](http://www.uniprot.org/uniprot/O54408) is the main one, but also [yjbM](http://genolist.pasteur.fr/SubtiList/genome.cgi?gene_detail+BG13142), [ywaC](http://www.uniprot.org/uniprot/P39583) - small alarmone synthetases (SAS). Nanamiya et al (08)

*  [CodY](http://www.uniprot.org/uniprot/P39779) represses fla/che and [hag](http://www.uniprot.org/uniprot/P02968). codY upstream of fla/che. (Bergana et al 03)

*  [DegU](http://www.uniprot.org/uniprot/P13800) enhances fla/che (Tsukahara & Ogura, 08) but DegU-P represses (Amati et al, 04)

*  [AmyE](http://www.uniprot.org/uniprot/P00691) next to origin of rep - therefore most cells have 2 copies. fla/che is near terminus, so typically just one copy (Yang et al, 99).

*  the flagellar pump [flgM](http://www.uniprot.org/uniprot/P39809) out of cell. flgM activates sigD? +ve feedback. (think about this!)
    * find mutants which have no basal body, and hence can't secrete flgM. check literature...

*  **stringent response** - transcription is blocked. would expect nucleoids to condense. check with DAPI.

















































































### Papers


*  Amati et al (04) [DOI](http://dx.doi.org/10.1128/JB.186.18.6003–6014.2004) (doesn't work) [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/15342569)
    * DegU-P represses fla/che. Also, codY upstream of fla/che - transcriptional readthrough, etc.
    * Interactions: DegU-P -| fla/che
    * Interesting strains: PB5368: **Pspac codY (with no terminator, readthrough to) fla/che** - eg. inducible


*  Bergana et al (03) [DOI](http://dx.doi.org/10.1128/JB.185.10.3118-3126.2003)
    * CodY a nutritional repressor of flagellar gene expression. The hag gene is repressed by CodY in nutrient rich environments. CodY binds to both Phag and the fla/che promoter to repress transcription. CodY triggered by AAs. 
    * Interactions: AAs -> CodY, GTP -> CodY, CodY -| fla/che, CodY -| hag
    * Interesting strains: hag-lacZ, codY146::cat


*  Breier and Grossman (09) [DOI](http://dx.doi.org/10.1128/JB.01294-08)
    * Dynamic Association of the Replication Initiator and Transcription Factor DnaA with the Bacillus subtilis Chromosome during Replication Stress
    * Link with DnaA? Not really sure how this fits in. Get a copy of full paper.
    * Interactions: ?
    * Interesting strains: ?


*  Calvio et al (08) [DOI](http://dx.doi.org/10.1128/JB.00455-08)
    * Autoregulation of swrAA and Motility in Bacillus. Positive feedback for swrA. Need to read properly to understand this better.
    * Interactions: sigA -> swrAA, sigD -> swrAA, swrAA -> sigD 
    * Interesting strains: various swrAA/degU type mutants


*  Caramori et al (96) No DOI? [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/8655488)
    * FlgM and SigD. FlgM is an anti-sigD?
    * Interactions: FlgM -| sigD
    * Interesting strains: various lacZ promoter reporters


*  Chen et al (09) [DOI](http://dx.doi.org/10.1128/JB.00521-09)
    * sigmaD, lytic enzymes, cell separation, motile sub-population, etc.


*  Eymann et al (02) [DOI](http://dx.doi.org/10.1128/JB.184.9.2500-2520.2002)
    * Key paper on stringent response and relA. Proteomic and transciptomic screening. Map of protein-encoding genes affected by (p)ppGpp. relA dependent induced genes include spo0A, spo0F, sog, etc, and flgM.
    * Interactions: relA -> flgM (indirect), relA -> spo0A (indirect), etc.


*  Ferullo and Lovett (08) [DOI](http://dx.doi.org/10.1371/journal.pgen.1000300)
    * E Coli. Stringent response and nucleoid localisation. In cells which cannot synthesize ppGpp, nucleoids are highly condensed. Would like to check this in Bacillus.


*  Kearns and Losick (05) [DOI](http://dx.doi.org/10.1101/gad.1373905). 
    * Key paper on motility. The fla/che operon is activated by both sigA and SwrA. It encodes various flagellum/chemotaxis genes (31 genes in operon), and also sigD and swrB. sidD and swrB activate various flagellum biosynthetic genes including Phag/hag and PflgM/flgM. Use Phag-gfp as reporter for SigD, as it is strongly activated by SigD. Also use a Phy-spank-swrA fusion for IPTG inductionn of fla/che operon.
    * Interactions: sigA -> fla/che, SwrA -> fla/che (inc. sigD, swrB), sigD -> flgM, SwrB -> flgM, sigD -> hag, SwrB -> hag (and flhF too...)
    * Interesting strains: amyE::Phag-gfp, Phy-spank-swrA, amyE::Pfla/che-lacZ


*  Kobayashi (07) [DOI](http://dx.doi.org/10.1111/j.1365-2958.2007.05923.x)
    * degU and how that fits in to the motility story. DegU binds to P(fla/che). But degU mutants are not motile. Stuff about high/low concentrations of DegU - need to read this properly to figure out what is going on.
    * Interactions: DegU -| fla/che (but read paper)
    * Interesting strains: degU::cat, **amyE::flgB P(A)-gfp**, and various related strains.


*  Krasny et al (08) [DOI](http://dx.doi.org/10.1111/j.1365-2958.2008.06256.x)
    * Very interesting paper on stringent response, ppGpp and ATP/GTP. Suggests that in Bacillus (as distinct from E. Coli) ppGpp mainly acts indirectly by suppressing GTP. Promoters sensitive to decrease in GTP levels. Possible mechanism outlined
    * Interactions: relA -> ppGpp, ppGpp -| GTP, ppGpp -> ATP, GTP -| sigA genes
    * Interesting strains: various promoter-lacZ fusions


*  Krasny and Gourse (04) [DOI](http://dx.doi.org/10.1038/sj.emboj.7600423)
    * rRNA transcription regulation and ribosome biosynthesis. Regulation of rrnB and rrnO rRNA promoters. GTP is the NTP initiating transcription from rRNA promoters.
    * Interactions: GTP -> rrnB, GTP -> rrnO. eg. GTP activates ribosome biosynthesis.
    * Interesting strains: have to check supplementary material


*  Lewis et al (00) no DOI? [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/10675340)
    * Compartmentalization of transcription and translation in Bacillus subtilis
    * stuff about nucleoids, stringent response, dapi staining, etc.


*  Liu and Zuber (98) No DOI? [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/9696775)
    * A Molecular Switch Controlling Competence and Motility: Competence Regulatory Factors ComS, MecA, and ComK Control sigD-Dependent Gene Expression in Bacillus subtilis
    * Possibly misleading to look for bistability when growing in minimal (=competence) medium, as could be due to competence...


*  Lulko, Veening et al (07) [DOI](http://dx.doi.org/10.1128/AEM.00472-07)
    * Production and Secretion Stress Caused by Overexpression of Heterologous {alpha}-Amylase Leads to Inhibition of Sporulation and a Prolonged Motile Phase in Bacillus subtilis
    * Main interest right now is that it describes the **hag-gfp** strain JWV has given me.


*  Nanamiya et al (08) [DOI](http://dx.doi.org/10.1111/j.1365-2958.2007.06018.x) (doesn't work) [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/18067544)
    * In E. Coli, RelA and SpoT homologous (p)ppGpp synthetases. In Bacillus, just RelA? Two genes, yjbM and ywaC encoding a novel type of alarmone synthetase - small alarmone synthetases (SAS) - mode of action distinct from RelA-SpoT homologues
    * Interactions: yjbM -> ppGpp, ywaC -> ppGpp
    * Interesting strains: aprE::relA-Pspac, and various other **relA, yjbM and ywaC** mutants - we have some of these


*  Serizawa et al (04) [DOI](http://dx.doi.org/10.1016/j.gene.2003.12.024)
    * Systematic analysis of SigD-regulated genes in Bacillus subtilis by DNA microarray and Northern blotting analyses


*  Sureka et al (08) [DOI](http://dx.doi.org/10.1371/journal.pone.0001771)
    * Mycobacteria. But a really nice paper on feedback, noise, modelling, bistability, rel and the stringent response. Would be nice to do something like this for Bacillus.


*  Tsukahara & Ogura (08) [DOI](http://dx.doi.org/doi:10.1186/1471-2180-8-8)
    * Promoter selectivity of the Bacillus subtilis response regulator DegU, a positive regulator of the fla/che operon and sacB. Unphosphorylated DegU positively regulates the fla/che operon. Get a full copy of the paper.
    * Interactions: DegU -> fla/che
    * Interesting strains: ?

*  Tsukahara & Ogura (08) [DOI](http://dx.doi.org/10.1111/j.1574-6968.2007.01019.x)
    * Characterization of DegU-dependent expression of bpr in Bacillus subtilis


*  Veening et al. (08) [DOI](http://dx.doi.org/10.1038/msb.2008.18)
    * Transient heterogeneity in extracellular protease production by Bacillus subtilis
    * Describes JWV's inducible DegUhy strain
    * Interesting strains: **inducible DegUhy**

*  Wendrich & Marahiel (03) [DOI](http://dx.doi.org/10.1046/j.1365-2958.1997.5511919.x)
    * Cloning and characterization of a relA/spoT homologue from Bacillus subtilis
    * Interesting strains: TW20 - **inducible relA** - we have this strain


*  West et al (00) No DOI? [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/10940026)
    * Motility regulation in Bacillus. fla/che PA, PD-3, and PsigD, etc. Three promoters for sigD. Two for fla/che and one specific to sigD. fla/che P(A) most important. the P(D) promoter doesn't work as a positive feedback loop.
    * **Important paper - read properly!**
    * Interactions: sigA -> fla/che
    * Interesting strains: various fla/che promotor mutations (deletions), but no gfp tags, and doesn't appear to be anything inducible.


*  Yang et al (99) No DOI? [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/10217790)
    * SigD not enough to trigger motility. FlgM is an anti-sigma factor (for sigF, in E. Coli, at least).
    * Interactions: FlgM -| sigD?
    * Interesting strains: amyE::(cat sigD-lacZ), sigD (N-term)-lacZ


*  Zanen et al (04) [DOI](http://dx.doi.org/10.1128/JB.186.17.5956–5960.2004) (doesn't work) [Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/15317803)
    * FlhF is a GTP-ase in the fla/che operon. Unlike other two signal recognition particle GTP-ases (Ffh and FtsY), FlhF is dispensible for protein secretion. Also inducible Pspac in fla/che at ylxH for control of genes downstream of ylxH.
    * Interactions: FlhF -| GTP
    * Interesting strains: **flhF::cat** (got it) and **ylxH::pMutin2** (BFA2616 - got it)






 --- //[Darren Wilkinson](people/darren) 18/1/09 //




